### PR TITLE
refactor: refactor region lease keeper

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

I realized we can't simply retain active leader regions and also need to find out the downgraded leader regions.

1. Refactor `find_staled_leader_regions`; now, it can find out the downgraded leader regions.
2. Add `find_staled_follower_regions`, returning upgradable and closable regions.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#2700

### CODE REVIEW

**Review by MichaelScofield — APPROVED** [review]

File: src/meta-srv/src/region/lease_keeper.rs [hunk 1]
```diff
@@ -59,33 +48,103 @@ impl RegionLeaseKeeper {
...
+        &self,
+        table_ids: &[TableId],
+    ) -> Result<HashMap<TableId, TableRouteValue>> {
+        let table_route_manager = self.table_metadata_manager.table_route_manager();
 
         // The subset of all table metadata.
         // TODO: considers storing all active regions in meta's memory.
         let metadata_subset = table_route_manager
-            .batch_get(&table_ids)
+            .batch_get(table_ids)
```
Comment (fengjiachun) [inline src/meta-srv/src/region/lease_keeper.rs:63]: How many tables' metadata will be loaded at once?

Comment (WenyXu) [reply src/meta-srv/src/region/lease_keeper.rs:63]: Typically, the number is the all leader(follower) regions at once. We might need to maintain a small piece of metadata in the leader's memory to track all expected/opening regions of each datanodes. No more worry it doesn't hit the cache.

**Comment by WenyXu** [comment]
@fengjiachun PTAL

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**